dotnet/src/Connectors/VectorData.Abstractions/RecordDefinition/VectorStoreRecordDefinition.cs

```diff
@@ -13,16 +14,20 @@ namespace Microsoft.Extensions.VectorData;
 /// </remarks>
 public sealed class VectorStoreRecordDefinition
 {
-    /// <summary>Empty static list for initialization purposes.</summary>
-    private static readonly List<VectorStoreProperty> s_emptyFields = new();
+    private IList<VectorStoreProperty>? _properties;
 
     /// <summary>
     /// Gets or sets the list of properties that are stored in the record.
     /// </summary>
-    public IReadOnlyList<VectorStoreProperty> Properties { get; init; } = s_emptyFields;
+    [AllowNull]
+    public IList<VectorStoreProperty> Properties
```

Comment:
I still believe that we should expose just the list rather than an ability to provide your own collection. Anyway, it could be just a `List` here as we don't really benefit from exposing an abstraction (in case of `IReadOnlyList` it could be an array)

Reply:
I mainly did `IList<T>` to align with MEAI - [see this comment](https://github.com/microsoft/semantic-kernel/pull/11882#discussion_r2072651791). I don't really have strong feelings on this - if the user happens to have e.g. an array, IList allows that to be assigned directly (whereas List does not). Any specific thoughts/arguments for exposing List instead of IList?

(will go ahead and merge to move forward, but let's continue the conversation and I can always change it later)